packages/fxa-content-server/app/scripts/views/settings/change_password.jsx

```diff
@@ -0,0 +1,319 @@
+import Translator from '../../lib/translator';
+
+const translator = new Translator();
+
+const t = msg => translator.get(msg);
+
+function ChangePassword(props) {
+ return (
+  <div id="change-password" className="settings-unit">
+   <ChangePasswordHeader />
+   <ChangePasswordForm
+      account={props.account}
+      submit={props.submit}
+      displayError={props.displayError}
+      isPanelOpen={props.isPanelOpen}
+      showValidationError={props.showValidationError}
+   />
+  </div>
+ );
+}
+
+export function ChangePasswordHeader() {
+ return (
+  <div className="settings-unit-stub">
+   <header className="settings-unit-summary">
+    <h2 className="settings-unit-title">{t('Password')}</h2>
+   </header>
+   <button
+    className="settings-button secondary-button settings-unit-toggle"
+    data-href="settings/change_password"
+   >
+    {t('Change...')}
+   </button>
+  </div>
+ );
+}
+
+export class ChangePasswordForm extends React.Component {
+ constructor(props) {
+  super(props);
+
+  this.state = {
+   hasFocused: false,
+   mail: props.account.get('email') || '',
+   newPass: '',
+   newVPass: '',
+   oldPass: '',
+  };
+ }
+ setOldPassword = event => {
+  this.setState({
+   oldPass: event.target.value,
+  });
+ };
+
+ setNewPassword = event => {
+  this.setState(
+   {
+    newPass: event.target.value,
+   });
+ };
+
+ setNewVPassword = event => {
+  this.setState(
+   {
+    newVPass: event.target.value,
+   });
+ };
+ showValidationErrorsEnd() {
+  if (this.state.newPass && this.validateFormField('#new_password') && this.state.newVPass && this.state.newPass !== this.state.newVPass) {
+   const err = AuthErrors.toError('PASSWORDS_DO_NOT_MATCH');
+   this.props.showValidationError('#new_vpassword', err);
+  }
+ }
+ componentDidUpdate() {
+  if (!this.state.hasFocused && this.props.isPanelOpen) {
+   this._input.focus();
+   this.setState({ hasFocused: true });
+  }
+  this.showValidationErrorsEnd();
+ }
+ componentWillUnmount() {
+  return this.state.newPass === this.state.newVPass;
+ }
+
+ setFocus = () => {
+  this.setState(
+   {
+    newPass: '',
+    newVPass: '',
+    oldPass: '',
+    hasFocused: false,
+   },
+   () => {
+    this.componentDidUpdate();
+   }
+  );
+ };
+
+ validateFormField(selector) {
+  try {
+   $(selector).validate();
+  } catch (err) {
+   this.props.showValidationError(selector, err);
+   return false;
+  }
+  return true;
+ }
+ handleSubmit = event => {
+  event.preventDefault();
+  const areInputsValid =
+   this.validateFormField('#old_password') &&
+   this.validateFormField('#new_password') &&
+   this.validateFormField('#new_vpassword');
+
+  if (!areInputsValid) {
+   return false;
+  }
+  if (this.state.newPass !== this.state.newVPass) {
+   const err = AuthErrors.toError('PASSWORDS_DO_NOT_MATCH');
+   this.props.showValidationError('#new_vpassword', err);
+  } else {
+   this.props.submit(this.state.oldPass, this.state.newPass).then(
+    () => {
+     this.setState({
```

Comment:
You might be able to call `this.setFocus()` here instead of duplicating the code.